drivers/include/soft_spi.h

```diff
@@ -0,0 +1,286 @@
+ * again.
+ *
+ * @param[in] bus       SPI device to release
+ */
+void soft_spi_release(soft_spi_t bus);
+
+/**
+ * @brief Transfer one byte on the given SPI bus
+ *
+ * @param[in] bus       SPI device to use
+ * @param[in] cs        chip select pin/line to use, set to SOFT_SPI_CS_UNDEF if chip
+ *                      select should not be handled by the SPI driver
+ * @param[in] cont      if true, keep device selected after transfer
+ * @param[in] out       byte to send out, set NULL if only receiving
+ *
+ * @return              the received byte
+ */
+uint8_t soft_spi_transfer_byte(soft_spi_t bus, soft_spi_cs_t cs, bool cont, uint8_t out);
+
+/**
+ * @brief   Transfer a number bytes using the given SPI bus
+ *
+ * @param[in]  bus      SPI device to use
+ * @param[in]  cs       chip select pin/line to use, set to SOFT_SPI_CS_UNDEF if chip
+ *                      select should not be handled by the SPI driver
+ * @param[in]  cont     if true, keep device selected after transfer
+ * @param[in]  out      buffer to send data from, set NULL if only receiving
+ * @param[out] in       buffer to read into, set NULL if only sending
+ * @param[in]  len      number of bytes to transfer
+ */
+void soft_spi_transfer_bytes(soft_spi_t bus, soft_spi_cs_t cs, bool cont,
+                        const void *out, void *in, size_t len);
+
+/**
+ * @brief   Transfer one byte to/from a given register address
+ *
+ * This function is a shortcut function for easier handling of SPI devices that
+ * implement a register based access scheme.
+ *
+ * @param[in] bus       SPI device to use
+ * @param[in]  cs       chip select pin/line to use, set to SOFT_SPI_CS_UNDEF if chip
+ *                      select should not be handled by the SPI driver
+ * @param[in] reg       register address to transfer data to/from
+ * @param[in] out       byte to send, set NULL if only receiving data
+ *
+ * @return              value that was read from the given register address
+ */
+uint8_t soft_spi_transfer_reg(soft_spi_t bus, soft_spi_cs_t cs, uint8_t reg, uint8_t out);
+
+/**
+ * @brief   Transfer a number of bytes to/from a given register address
+ *
+ * This function is a shortcut function for easier handling of SPI devices that
+ * implement a register based access scheme.
+ *
+ * @param[in]  bus      SPI device to use
+ * @param[in]  cs       chip select pin/line to use, set to SOFT_SPI_CS_UNDEF if chip
+ *                      select should not be handled by the SPI driver
+ * @param[in]  reg      register address to transfer data to/from
+ * @param[in]  out      buffer to send data from, set NULL if only receiving
+ * @param[out] in       buffer to read into, set NULL if only sending
+ * @param[in]  len      number of bytes to transfer
+ */
+void soft_spi_transfer_regs(soft_spi_t bus, soft_spi_cs_t cs, uint8_t reg,
+                       const void *out, void *in, size_t len);
+
+#ifdef __cplusplus
+}
+#endif
+
+#endif /* SOFT_SOFT_SPI_H */
+/** @} */
```

Comment:
Please check the whole documentation of this file. It can be drastically reduced. One thing that I missed is the hint about ignored clock speeds in the "module description" at the beginning of this file. Remember that you want to explain the soft SPI driver here, not any generic or hardware dependent interface

Reply:
:heavy_check_mark: